lighthouse-core/report/html/renderer/report-ui-features.js

```diff
@@ -425,6 +439,32 @@ class ReportUIFeatures {
         this._toggleDarkTheme();
         break;
       }
+      case 'swap-locale': {
+        if (!this._swapLocaleOptions) throw new Error('must call .initSwapLocale first');
+
+        // TODO: esmodules would be nice here...
+        // @ts-ignore
+        if (!window.Lighthouse || !window.Lighthouse.i18n) await import('./i18n-module.js');
+
+        // Waiting for UI :)
+        const randomLocales = ['es', 'fr', 'vi', 'en-US'];
+        window.__locale_index = window.__locale_index || 0;
+        const locale = randomLocales[window.__locale_index++ % randomLocales.length];
+
+        // TODO: locales.js maps a locale code to a locale module. Need that same mapping,
+        // but for locale code to locale module _name_, so that we can fetch w/ that same
+        // name here.
+        const localModuleName = locale;
+        const lhlMessages = await this._swapLocaleOptions.fetchData(localModuleName);
+        if (!lhlMessages) throw new Error(`could not fetch data for locale: ${locale}`);
+
+        window.Lighthouse.i18n.registerLocaleData(locale, lhlMessages);
+        const newLhr = window.Lighthouse.i18n.swapLocale(this.json, locale).lhr;
+        this._swapLocaleOptions.render(newLhr);
```

Comment:
I don't love this indirection - any better ideas?

Reply:
yeah don't love this either, but a trigger to `refresh` seems reasonable for the UI features class to have.
 

a slightly cleaner way I'm thinking of would be this class emitting a `refresh` event with the new LHR that the report viewer can listen to and reinvoke `render` + `initFeatures`?

Reply:
done